lib/fastlane/plugin/revenuecat_internal/actions/scan_with_flaky_test_retries.rb

```diff
@@ -0,0 +1,261 @@
+require 'nokogiri'
+require 'scan'
+require 'fastlane_core/ui/ui'
+
+module Fastlane
+  UI = FastlaneCore::UI unless Fastlane.const_defined?(:UI)
+
+  module Actions
+    class ScanWithFlakyTestRetriesAction < Action
+      def self.run(params)
+        # Pulling out the values from Fastlane::Configurations to
+        # modify and pass into `scan`
+        param_values = params.values
+
+        # Delete option that isn't part of scan
+        number_of_flaky_retries = param_values.delete(:number_of_flaky_retries)
+
+        if number_of_flaky_retries == 0
+          # Run scan directly if no retries are needed
+          other_action.scan(**param_values)
+        else
+          # Delete option that is part of scan but we will replace
+          output_directory = param_values.delete(:output_directory)
+
+          # Storing all scan output in a separate temp directory
+          begin
+            temp_dir = Dir.mktmpdir
```

Comment:
Nit: It doesn't look like this directory is ever cleaned up. I _assume_ that CircleCI will probably delete it when the job is finished and it destroys the image. 

Would it be a good idea to delete it when we're done in case this ever gets run on someone's local machine or in a CI environment that persists files across CI runs?